demo/react/src/VoiceWidget.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useCallback, useEffect, useState, useRef } from "react";
+import { useCheetah } from "@picovoice/cheetah-react";
+
+import cheetahModel from "./lib/cheetahModel";
+
+export default function VoiceWidget() {
+  const accessKeyRef = useRef<string>("");
+
+  const [isBusy, setIsBusy] = useState<boolean>(false)
+  const [transcript, setTranscript] = useState<string>("");
+
+  const { result, isLoaded, isListening, error, init, start, stop, release } =
+    useCheetah();
+
+  useEffect(() => {
+    if (result !== null) {
+      setTranscript((prev) => {
+        let newTranscript = prev + result.partialTranscript;
+
+        if (result.isComplete) {
+          newTranscript += " ";
+        }
+
+        return newTranscript;
+      });
+    }
+  }, [result]);
+
+  const initEngine = useCallback(async () => {
+    if (accessKeyRef.current.length === 0) {
+      return;
+    }
+
+    setIsBusy(true)
+    await init(accessKeyRef.current, cheetahModel, {
+      enableAutomaticPunctuation: true,
+    });
+    setIsBusy(false)
+  }, [init]);
+
+  const toggleRecord = async () => {
+    setIsBusy(true)
+    if (isListening) {
+      await stop();
+    } else {
+      await start();
```

Comment:
I think it would be good to clear previous results when starting a new recording